sycl/doc/SYCL_compiler_and_runtime_design.md

```diff
@@ -42,6 +42,91 @@ back-end. Today middle-end transformations include just a couple of passes:
 - **Back-end** - produces native "device" code in ahead-of-time compilation
 mode.
 
+### SYCL support in clang frontend
+
+SYCL support in clang frontend can be split into the following components:
+
+- Device code outlining. Since SYCL is a single source programming model
+compiler should be able to separate device code from host code.
+
+- Lowering of lambda function objects and named function objects ("SYCL kernel
+functions"). To execute "SYCL kernel functions" on OpenCL devices some
+transformations are required.
+
+- Device code diagnostics.
+
+- Integration header generation.
+
+#### Device code outlining
+
+Here is a code example of a SYCL program that demonstrates compiler outlining
+work:
+
+```C++
+int foo(int x) { return ++x; }
+int bar(int x) { throw std::exception("CPU code only!"); }
+...
+using namespace cl::sycl;
+queue Q;
+buffer<int, 1> a(range<1>{1024});
+Q.submit([&](handler& cgh) {
+      auto A = a.get_access<access::mode::write>(cgh);
+      cgh.parallel_for<init_a>(range<1>{1024}, [=](id<1> index) {
+        A[index] = index[0] * 2 + index[1] + foo(42);
+      });
+    }
+...
+```
+
+SYCL compiler needs to compile lambda exression passed to
+`cl::sycl::handler::parallel_for` method and function `foo` called from this
+lambda function. Compiler also must ignore bar function when we compile the "device" part
+of the single source code.
+
+Current approach is to use the SYCL kernel atttribute in SYCL runtime to mark code
```

Comment:
Current -> The current 
in SYCL runtime -> in the SYCL runtime
Obviously runtime -> The SYCL runtime

Reply:
Done, thanks!